ab2d-bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDSearchImplTest.java

```diff
@@ -0,0 +1,137 @@
+import gov.cms.ab2d.fhir.FhirVersion;
+
+class BFDSearchImplTest {
+
+  private static int patientId = 1;
+  private static OffsetDateTime since = OffsetDateTime.now();
+  private static OffsetDateTime until = OffsetDateTime.now();
+  private static int pageSize = 10;
+  private static String bulkJobId = "bulkJobId";
+  private static FhirVersion version = FhirVersion.R4;
+  private static String contractNum = "contractNum";
+  private static String filePath = "src/test/resources/bb-test-data/eob/20010000001115.json";
+  private static String activeProfile = "test";
+  private static String bfdUrl = "http://localhost:8080";
+
+  HttpClient httpClient200() throws IOException {
+    HttpClient httpClient = mock(HttpClient.class);
+    CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+    when(httpClient.execute(any())).thenReturn(response);
+    when(response.getStatusLine()).thenReturn(mock(org.apache.http.StatusLine.class));
+    when(response.getStatusLine().getStatusCode()).thenReturn(HttpStatus.SC_OK);
+    when(response.getEntity()).thenReturn(mock(org.apache.http.HttpEntity.class));
+    String fileContent = Files.readString(Paths.get(filePath));
+    when(response.getEntity().getContent()).thenReturn(new ByteArrayInputStream(fileContent.getBytes()));
+    return httpClient;
+  }
+
+  HttpClient httpClient404() throws IOException {
+    HttpClient httpClient = mock(HttpClient.class);
+    CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+    when(httpClient.execute(any())).thenReturn(response);
+    when(response.getStatusLine()).thenReturn(mock(org.apache.http.StatusLine.class));
+    when(response.getStatusLine().getStatusCode()).thenReturn(HttpStatus.SC_NOT_FOUND);
+    return httpClient;
+  }
+
+  HttpClient httpClient500() throws IOException {
+    HttpClient httpClient = mock(HttpClient.class);
+    CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+    when(httpClient.execute(any())).thenReturn(response);
+    when(response.getStatusLine()).thenReturn(mock(org.apache.http.StatusLine.class));
+    when(response.getStatusLine().getStatusCode()).thenReturn(HttpStatus.SC_INTERNAL_SERVER_ERROR);
+    return httpClient;
+  }
+
+  @Test
+  void testSearchEOB200() throws IOException{
+    // Setup mocks
+    HttpClient httpClient = httpClient200();
+    Environment environment = mock(Environment.class);
+    when(environment.getActiveProfiles()).thenReturn(new String[] { activeProfile });
+
+    // Setup classes
+    BfdClientVersions bfdClientVersions = new BfdClientVersions(bfdUrl, httpClient);
+    BFDSearchImpl bfdSearchImpl = new BFDSearchImpl(httpClient, environment, bfdClientVersions);
+
+    // Business logic & assertion(s)
+    IBaseBundle result = bfdSearchImpl.searchEOB(patientId, since, until, pageSize, bulkJobId, version, contractNum);
+    assertNotNull(result);
+  }
+
+  @Test
+  void testSearchEOB200WithFalsyArgs() throws IOException{
```

Comment:
The purpose of this test is just to get code coverage on all of the `if` statements in [`BFDSearchImpl`](https://github.com/CMSgov/AB2D-Libs/blob/main/ab2d-bfd/src/main/java/gov/cms/ab2d/bfd/client/BFDSearchImpl.java)